src/poetry/core/vcs/git.py

```diff
@@ -11,31 +11,46 @@
 
 
 PROTOCOL = r"\w+"
-USER = r"[a-zA-Z0-9_.-]+"
+URL_RESTRICTED = r"[^/\?#:@]"
```

Comment:
LGTM. I just wonder if there is a reference for this / if this is complete or if it is just a best guess (which might also be ok)?

Reply:
We could use https://url.spec.whatwg.org/#forbidden-host-code-point.  This one is also tricky as the value may not be b64 encoded at the point of parsing.

Reply:
Maybe. (I like to have a comment with a link. 😀) But we can also keep it as is if you think it does not make sense. I leave the decision to you.